pliers/extractors/models.py

```diff
@@ -1,14 +1,193 @@
 ''' Extractor classes based on pre-trained models. '''
 
 import numpy as np
+import pandas as pd
 
 from pliers.extractors.image import ImageExtractor
-from pliers.extractors.base import ExtractorResult
+from pliers.extractors.base import Extractor, ExtractorResult
 from pliers.filters.image import ImageResizingFilter
-from pliers.utils import attempt_to_import, verify_dependencies
+from pliers.stimuli import ImageStim, TextStim, AudioStim
+from pliers.utils import (attempt_to_import, verify_dependencies,
+                         listify)
+
+import logging
 
 tf = attempt_to_import('tensorflow')
+hub = attempt_to_import('tensorflow_hub')
 attempt_to_import('tensorflow.keras')
+attempt_to_import('tensorflow_text')
+
+
+class TFHubExtractor(Extractor):
+
+    ''' A generic class for Tensorflow Hub extractors 
+    Args:
+        url_or_path (str): url or path to TFHub model. You can
+            browse models at https://tfhub.dev/.
+        task (str): model task/domain identifier
+        features (optional): list of labels (for classification) 
+            or other feature names. If a one-element list or 
+            a string is passed, packs output vectors into one column.
+            If not specified, returns numbered features 
+            (feature_0, feature_1, ... ,feature_n).
+        kwargs (dict): arguments to hub.KerasLayer call
+    '''
+
+    _log_attributes = ('url_or_path', 'features', 'transform_out')
+
+    def __init__(self, url_or_path, features=None, task=None, 
+                 transform_out=None, **kwargs):
+        verify_dependencies(['tensorflow', 'tensorflow_hub', 
```

Comment:
Are all of these dependencies needed here? At least from the code, it looks like it might only be `tensorflow_hub` (for `KerasLayer`).

Reply:
only hub is required, indeed.
Removed tensorflow dependency, and moved tensorflow_text to the text extractor.
It is only needed for some models and there's no way to know if it is needed until the model is called, so I've added a warning at initialization when import fails.